lib/ae_mdw/oracles.ex

```diff
@@ -60,6 +65,59 @@ defmodule AeMdw.Oracles do
     end
   end
 
+  @spec fetch_oracle_queries(state(), pubkey(), pagination(), range(), cursor() | nil) ::
+          {:ok, {cursor() | nil, [oracle_query()], cursor() | nil}} | {:error, Error.t()}
+  def fetch_oracle_queries(state, oracle_id, pagination, nil, cursor) do
+    with {:ok, oracle_pk} <- Validate.id(oracle_id, [:oracle_pubkey]) do
+      cursor = deserialize_queries_cursor(cursor, oracle_pk)
+      key_boundary = {{oracle_pk, Util.min_bin()}, {oracle_pk, Util.max_256bit_bin()}}
+
+      {prev_cursor, query_ids, next_cursor} =
+        fn direction ->
+          Collection.stream(state, @table_query, direction, key_boundary, cursor)
+        end
+        |> Collection.paginate(pagination)
+
+      queries = Enum.map(query_ids, &render_query(state, &1))
+
+      {:ok,
+       {serialize_queries_cursor(prev_cursor), queries, serialize_queries_cursor(next_cursor)}}
+    end
+  end
+
+  def fetch_oracle_queries(_state, _oracle_id, _pagination, _range, _cursor),
+    do: {:error, ErrInput.Query.exception(value: "cannot filter by range on this endpoint")}
+
+  defp deserialize_queries_cursor(nil, _oracle_pk), do: nil
+
+  defp deserialize_queries_cursor(query_id, oracle_pk) do
+    case Enc.safe_decode(:oracle_query_id, query_id) do
+      {:ok, query_id} -> {oracle_pk, query_id}
+      {:error, _reason} -> nil
```

Comment:
I think we should return invalid cursor error here.

Reply:
Good call, done